internal/controller/cilium_flow_collector.go

```diff
@@ -26,20 +29,28 @@ type CiliumFlowCollector struct {
 const (
 	ciliumHubbleRelayMaxFlowCount uint64 = 100
 	ciliumHubbleRelayServiceName  string = "hubble-relay"
+
+	// Constants for fetching the mTLS secret
+	hubbleMTLSSecretName       string = "hubble-relay-client-certs"
+	ciliumHubbleRelayNamespace string = "kube-system"
 )
 
 var (
 	ErrHubbleNotFound   = errors.New("hubble Relay service not found; disabling Cilium flow collection")
 	ErrNoPortsAvailable = errors.New("hubble Relay service has no ports; disabling Cilium flow collection")
 )
 
-// discoverCiliumHubbleRelayAddress uses a kubernetes clientset in order to discover the address of the hubble-relay service within kube-system.
-func discoverCiliumHubbleRelayAddress(ctx context.Context, ciliumNamespace string, clientset kubernetes.Interface) (string, error) {
+func discoverCiliumHubbleRelay(ctx context.Context, ciliumNamespace string, clientset kubernetes.Interface) (*v1.Service, error) {
 	service, err := clientset.CoreV1().Services(ciliumNamespace).Get(ctx, ciliumHubbleRelayServiceName, metav1.GetOptions{})
 	if err != nil {
-		return "", ErrHubbleNotFound
+		return nil, ErrHubbleNotFound
 	}
 
+	return service, nil
+}
+
+// retrieveHubbleRelayAddress uses a kubernetes clientset to discover the address of the hubble-relay service within kube-system.
+func retrieveHubbleRelayAddress(service *v1.Service) (string, error) {
```

Comment:
1. Fix this comment. It doesn't match the implementation.
1. retrieve -> get
```suggestion
// getHubbleRelayAddress returns the address of the hubble-relay service to connect a gRPC client to.
func getHubbleRelayAddress(service *v1.Service) (string, error) {
```